uaclient/cli.py

```diff
@@ -1072,36 +1038,17 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
             raise exceptions.UserFacingError(
                 ua_status.MESSAGE_DETACH_AUTOMATION_FAILURE
             )
-    contract_client = contract.UAContractClient(cfg)
-    try:
-        tokenResponse = contract_client.request_auto_attach_contract_token(
-            instance=instance
-        )
-    except contract.ContractAPIError as e:
-        if e.code and 400 <= e.code < 500:
-            raise exceptions.NonAutoAttachImageError(
-                ua_status.MESSAGE_UNSUPPORTED_AUTO_ATTACH
-            )
-        raise e
-    if current_iid:
-        cfg.write_cache("instance-id", current_iid)
-
-    return tokenResponse["contractToken"]
 
-
-@assert_root
-@assert_lock_file("ua auto-attach")
-def action_auto_attach(args, *, cfg):
-    disable_auto_attach = util.is_config_value_true(
-        config=cfg.cfg, path_to_value="features.disable_auto_attach"
-    )
-    if disable_auto_attach:
-        msg = "Skipping auto-attach. Config disable_auto_attach is set."
-        logging.debug(msg)
-        print(msg)
+    try:
+        actions.auto_attach(cfg, instance)
+    except util.UrlError:
+        print(ua_status.MESSAGE_ATTACH_FAILURE)
+        return 1
+    except Exception:
```

Comment:
I think we should inform the user about the `Exception` somehow. 
Even if it is an unknown `Exception`  we could log it and say to the user to check the logs

Reply:
You're totally right. I think we can just let it bubble up

Reply:
Oh I think I mean to catch `UserFacingError` here to match the old implementation line 982.

Reply:
@lucasmoura updated